### PR TITLE
feat: add Clickhouse

### PR DESCRIPTION
**Pull request by uebriges**
https://clickhouse.com/

### CODE REVIEW

File: clickhouse/clickhouse.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  echo -e "${CGreen}${1}${NC}"
+}
+
+# vars
+# os="${os:-${1}}"
+arch="${arch:-${2}}"
+version="${version:-${3}}"
+
+case $arch in
+amd64) arch_alias="";;
```
Comment (franklinkim) [inline clickhouse/clickhouse.sh]: set to `amd64` otherwise it will be empty further down

File: clickhouse/clickhouse.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+error() {
+  echo -e "${CRed}${1}${NC}"
+}
+
+success() {
+  echo -e "${CGreen}${1}${NC}"
+}
+
+# vars
+# os="${os:-${1}}"
```
Comment (franklinkim) [inline clickhouse/clickhouse.sh]: check os and warn & skip if `linux` for now, since there is no binary for that

File: clickhouse/clickhouse.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+case $arch in
+amd64) arch_alias="";;
+arm64) arch_alias="aarch64";;
+esac
+
+# https://github.com/ClickHouse/ClickHouse/releases/download/v24.4.1.2088-stable/clickhouse-macos
+# https://github.com/ClickHouse/ClickHouse/releases/download/v24.4.1.2088-stable/clickhouse-macos-aarch64
+
+url="https://github.com/ClickHouse/ClickHouse/releases/download/${version}/clickhouse-macos-${arch_alias}"
```
Comment (franklinkim) [inline clickhouse/clickhouse.sh]: use `os`

File: clickhouse/clickhouse.sh [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+
+info "downloading ..."
+echo $url
```
Comment (franklinkim) [inline clickhouse/clickhouse.sh]: no debug output for now

**Review by franklinkim — CHANGES_REQUESTED** [review]

File: clickhouse/clickhouse.sh (hunk 3 above)
Comment (uebriges) [reply clickhouse/clickhouse.sh]: os in my case is "darwin". That doesn't fit here. Do mean to hard coded it with "macos"?

Comment (franklinkim) [reply clickhouse/clickhouse.sh]: us `os_alias` similar to `arch_alias`. see other implementations

Comment (uebriges) [reply clickhouse/clickhouse.sh]: done

File: clickhouse/clickhouse.sh [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+case $arch in
+amd64) arch_alias="amd64";;
+arm64) arch_alias="aarch64";;
+esac
+
+# https://github.com/ClickHouse/ClickHouse/releases/download/v24.4.1.2088-stable/clickhouse-macos
+# https://github.com/ClickHouse/ClickHouse/releases/download/v24.4.1.2088-stable/clickhouse-macos-aarch64
+
+info "downloading ..."
+curl -fL "https://github.com/ClickHouse/ClickHouse/releases/download/${version}/clickhouse-${os_alias}-${arch_alias}" -o "${BIN_DIR}/clickhouse-${os_alias}-${version}-${arch}" --create-dirs
```
Comment (franklinkim) [inline clickhouse/clickhouse.sh]: output name is incorrect, don't use os_alias & order matters!

**Review by franklinkim — CHANGES_REQUESTED** [review]

**Review by franklinkim — APPROVED** [review]

**Pull request merged**